### PR TITLE
Fix basic_training.py

### PR DESCRIPTION
**Pull request by zzzwen**
loss needs grad

```
  File "/Users/wzz/work/torch-xla/torchax/test/.venv/lib/python3.12/site-packages/torch/autograd/graph.py", line 823, in _engine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
```

### CODE REVIEW

File: torchax/examples/basic_training.py [hunk 1]
```diff
@@ -133,6 +133,9 @@ def train_one_epoch(epoch_index, tb_writer=None):
 
     # Compute the loss and its gradients
     loss = loss_fn(outputs, labels)
+    # Manually set requires_grad to True
+    loss.requires_grad_(True)
```
Comment (yaoshiang) [inline torchax/examples/basic_training.py]: why is this necessary when it's not necessary in standard PT?

**Review by yaoshiang — CHANGES_REQUESTED** [review]

File: torchax/examples/basic_training.py (hunk 1 above)
Comment (zzzwen) [reply torchax/examples/basic_training.py]: Update the code, it is the output from the jax model that has requires_grad == False.
@qihqi, it will take me a while to figure out what's going on here, do you happen to know.

**Comment by yaoshiang** [comment]
At first blush, I'd assume this is a workaround for a bug in the internals. The requires_grad should be on for most of the model weights, triggering a grad on the loss value. Setting it after the criterion(expected, actual) already populates the grads. loss.requires_grad at that point should have no meaning. If this code change triggers behavior changes, I think there is something deeper at work here.

File: torchax/examples/basic_training.py (hunk 1 above)
Comment (qihqi) [reply torchax/examples/basic_training.py]: Yeah likely a bug.

**Comment by qihqi** [comment]
> At first blush, I'd assume this is a workaround for a bug in the internals. The requires_grad should be on for most of the model weights, triggering a grad on the loss value. Setting it after the criterion(expected, actual) already populates the grads. loss.requires_grad at that point should have no meaning. If this code change triggers behavior changes, I think there is something deeper at work here.

Yeah, likely the requires_grad (which should be propagated automatically) did not for some reason. We can fix that in a separate PR.

**Comment by zzzwen** [comment]
This wont work, the problem turn out to be a lot larger. the TL;DR is non of the torchax tensor has requires_grad, therefore autograd will not backprop any of these. 
Let' use the https://github.com/pytorch/xla/issues/9317 to track it
